tests/test_base_client.py

```
@@ -146,60 +147,346 @@ def client_contract(request):
+        def test_operation():
+            return "test"
+
+        with pytest.raises(AttributeError, match="must implement _execute_with_error_handling"):
+            client.execute_with_retry(test_operation, "test_op")
+
+    @patch('nui_lambda_shared_utils.base_client.get_secret')
+    @patch('nui_lambda_shared_utils.base_client.get_config')
+    @patch('nui_lambda_shared_utils.base_client.with_retry')
+    def test_execute_with_retry_configures_retry_decorator(self, mock_with_retry, mock_get_config, mock_get_secret):
+        """Test that retry decorator is configured correctly."""
+        mock_get_config.return_value = Mock(test_credentials_secret="test-secret")
```

Ruff (0.13.1)

Possible hardcoded password assigned to argument: "test_credentials_secret"

(S106)